app/frontend/actions.ts

```diff
@@ -11,32 +11,33 @@ import {
   mnemonicValidator,
   donationAmountValidator,
   poolIdValidator,
+  validatePoolRegUnsignedTx,
 } from './helpers/validators'
 import printAda from './helpers/printAda'
 import debugLog from './helpers/debugLog'
 import getConversionRates from './helpers/getConversionRates'
 import sleep from './helpers/sleep'
 import {
   NETWORKS,
-  PREMIUM_MEMBER_BALANCE_TRESHHOLD,
+  PREMIUM_MEMBER_BALANCE_TRESHOLD,
   BIG_DELEGATOR_THRESHOLD,
 } from './wallet/constants'
 import NamedError from './helpers/NamedError'
 import {exportWalletSecretDef} from './wallet/keypass-json'
-import {CardanoWallet} from './wallet/cardano-wallet'
 import mnemonicToWalletSecretDef from './wallet/helpers/mnemonicToWalletSecretDef'
 import sanitizeMnemonic from './helpers/sanitizeMnemonic'
 import {initialState} from './store'
 import {toCoins, toAda, roundWholeAdas} from './helpers/adaConverters'
 import captureBySentry from './helpers/captureBySentry'
-import {State, Ada, Lovelace} from './state'
-import CryptoProviderFactory from './wallet/byron/crypto-provider-factory'
+import {State, Ada, Lovelace, GetStateFn, SetStateFn, sourceAccountState} from './state'
 import ShelleyCryptoProviderFactory from './wallet/shelley/shelley-crypto-provider-factory'
-import {ShelleyWallet} from './wallet/shelley-wallet'
+import {Wallet} from './wallet/wallet'
+import {parseUnsignedTx} from './helpers/cliParser/parser'
+import {TxPlan, unsignedPoolTxToTxPlan} from './wallet/shelley/shelley-transaction-planner'
 import getDonationAddress from './helpers/getDonationAddress'
 import {localStorageVars} from './localStorage'
 
-let wallet: ReturnType<typeof CardanoWallet | typeof ShelleyWallet>
+let wallet: ReturnType<typeof Wallet>
```

Comment:
I liked ShelleyWallet more - it's easier to find in the code and pollutes the namespace to a lesser extent than a generic "Wallet" class which also leaves to much space for doubt about what kind of wallet it is

Reply:
I decided to drop this since I am planning to remove all byron related code so we would need to add "Shelley" keyword to everything. But I agree about the namespace, I ll rename it to shelley-wallet and we can think about this in a future refactor.